cc_plugin_glider/glider_dac.py

```diff
@@ -402,7 +402,9 @@ def check_depth_array(self, dataset):
         '''
         test_ctx = TestCtx(BaseCheck.HIGH, 'Profile data is valid')
         if 'depth' in dataset.variables:
-            test_ctx.assert_true(np.abs(np.diff(dataset.variables['depth']).sum()) > 1e-4,
+            depth = dataset.variables['depth'][:]
+            test_ctx.assert_true(np.abs(np.diff(depth[~depth.mask]).sum()) >
```

Comment:
Are we guaranteed to have a mask attribute?

Reply:
A fair question.  I'll generate a variable from CDL without a _FillValue defined and check.

Reply:
Looks like the answer is no, so I'll fix up the code.

```
[badams@localhost]~/devel/compliance-checker/compliance_checker/tests/data% ncdump timeseries.nc
netcdf timeseries {
dimensions:
        time = 2 ;
variables:
        double time(time) ;
                time:standard_name = "time" ;
                time:long_name = "Time" ;
                time:axis = "T" ;
                time:units = "seconds since 1970-01-01T00:00:00Z" ;
        float lat ;
                lat:standard_name = "latitude" ;
                lat:units = "degrees_north" ;
                lat:axis = "Y" ;
                lat:long_name = "Latitude" ;
        float lon ;
                lon:standard_name = "longitude" ;
                lon:units = "degrees_east" ;
                lon:axis = "X" ;
                lon:long_name = "Longitude" ;
        float z ;
                z:standard_name = "depth" ;
                z:positive = "down" ;
                z:units = "m" ;
                z:long_name = "Depth below surface" ;
        float temperature(time) ;
                temperature:standard_name = "sea_water_temperature" ;
                temperature:units = "deg_C" ;
                temperature:long_name = "Seawater Temperature" ;
                temperature:coordinates = "time lat lon z" ;
data:

 time = _, _ ;

 lat = _ ;

 lon = _ ;

 z = _ ;

 temperature = _, _ ;
}
[badams@localhost]~/devel/compliance-checker/compliance_checker/tests/data% ipython
Python 2.7.5 (default, Jul 13 2018, 13:06:57)
Type "copyright", "credits" or "license" for more information.

IPython 5.8.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from netCDF4 import Dataset

In [2]: ds = Dataset('timeseries.nc')

In [3]: ds.variables['temperature'][:]
Out[3]: array([  9.96920997e+36,   9.96920997e+36], dtype=float32)

In [4]: arr = ds.variables['temperature'][:]

In [5]: arr.mask
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-5-2dc3f87ab4fd> in <module>()
----> 1 arr.mask

AttributeError: 'numpy.ndarray' object has no attribute 'mask'
```

Reply:
Hold on, that might have been an old version of Python netCDF4.

Reply:
Upon modifying the CDL to actually have temperature data, it appears there are default _FillValues defined.  On netCDF4 >= 1.4.0, which is specified in compliance-checker's requirements.txt, the variables all seem to be `numpy.ma.MaskedArray` of the appropriate data type for the variable.  So I think this code should work fine, but please test against your own environment as a sanity check and ensure your netCDF4 version is greater than or equal to 1.4.0.

```
 dimensions:
        time = 2 ;
variables:
        double time(time) ;
                time:standard_name = "time" ;
                time:long_name = "Time" ;
                time:axis = "T" ;
                time:units = "seconds since 1970-01-01T00:00:00Z" ;
        float lat ;
                lat:standard_name = "latitude" ;
                lat:units = "degrees_north" ;
                lat:axis = "Y" ;
                lat:long_name = "Latitude" ;
        float lon ;
                lon:standard_name = "longitude" ;
                lon:units = "degrees_east" ;
                lon:axis = "X" ;
                lon:long_name = "Longitude" ;
        float z ;
                z:standard_name = "depth" ;
                z:positive = "down" ;
                z:units = "m" ;
                z:long_name = "Depth below surface" ;
        float temperature(time) ;
                temperature:standard_name = "sea_water_temperature" ;
                temperature:units = "deg_C" ;
                temperature:long_name = "Seawater Temperature" ;
                temperature:coordinates = "time lat lon z" ;
data:

 time = _, _ ;

 lat = _ ;

 lon = _ ;

 z = _ ;

 temperature = 3, 2.8 ;
}
(cchecker) [badams@localhost]~/devel/compliance-checker/compliance_checker/tests/data% ipython
Python 3.4.9 (default, Aug 14 2018, 21:28:57)
Type 'copyright', 'credits' or 'license' for more information
IPython 6.1.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from netCDF4 import Dataset

In [2]: ds = Dataset('timeseries.nc')

In [3]: ds.variables['temperature'][:]
Out[3]: 
masked_array(data = [3.0 2.799999952316284],
             mask = [False False],
       fill_value = 9.96921e+36)

In [4]: import netCDF4

In [5]: netCDF4
Out[5]: <module 'netCDF4' from '/home/badams/.virtualenvs/cchecker/lib/python3.4/site-packages/netCDF4/__init__.py'>

In [6]: netCDF4.__version__
Out[6]: '1.4.0'
```